### PR TITLE
fix: prevent execution from trying to get conditionForNextStatus for …

### PR DESCRIPTION
**Pull request by timigod**
prevents execution from trying to get conditionForNextStatus for waiting_pickup vehicle status

### CODE REVIEW

File: server/controllers/StatusController.js [hunk 1]
```diff
@@ -41,14 +41,16 @@ const getStatus = async (req, res) => {
...
       const status = 'in_mission';
-      const currentStatus = _.find(missionProgress, {'status': vehicle.status});
-      if (currentStatus.conditionForNextStatus(latestMission)){
-        const timestampString = currentStatus.nextMissionStatus + '_at';
-        let timestampObject = {};
-        timestampObject[timestampString] = Date.now();
-        await updateMission(latestMissionId, timestampObject);
-        await createMissionUpdate(latestMissionId, currentStatus.nextMissionStatus);
-        await updateVehicleStatus(latestMission.vehicle_id, currentStatus.nextVehicleStatus);
+      if (vehicle.status !== 'waiting_pickup'){
```
Comment (TalAter) [inline server/controllers/StatusController.js]: This seems like an unsustainable way to approach this.

You'll have to keep adding conditions here for when to skip the check as mission flows get more complicated.

You've created an elegant structure in `missionProgress.js` that defines the flow, and here you are going around it for one specific case. Why not handle this status transition within `missionProgress.js` just like you handle `available`

Comment (timigod) [reply server/controllers/StatusController.js]: Good idea!

**Comment by timigod** [comment]
@TalAter I have no idea why the build keeps failing. I've run `npm test` locally many times and all is well.

**Comment by TalAter** [comment]
@timigod 
The npm install is failing. Looks like a storm is brewing: https://github.com/npm/registry/issues/255

Things are about to get crazy

**Comment by TalAter** [comment]
It's working now.

**Pull request merged**